### ISSUE TITLE
STAR seg fault on small "genome"

### ISSUE THREAD

**Issue by JCSzamosi** [alexdobin/STAR#271]
I'm aligning reads to the human RepBase as follow:

index the genome:
```
STAR --runThreadN=10 
--runMode=genomeGenerate 
--genomeFastaFiles=$infast 
--genomeDir=$gendir 
--genomeSAindexNbases=9.2 # adjusted using the min(14,log2(size)/2-1) formula
```

map the reads:
```
STAR --runMode alignReads 
--runThreadN 10 
--genomeDir $gendir 
--genomeLoad LoadAndRemove 
--readFilesIn $f1 $f2 
--outSAMunmapped Within 
--outFilterMultimapNmax 30 
--outFilterMultimapScoreRange 1  
--outFileNamePrefix $res$bnm 
--outSAMattributes All 
--readFilesCommand zcat 
--outStd BAM_Unsorted 
--outSAMtype BAM Unsorted 
--outFilterType BySJout 
--outReadsUnmapped Fastx 
--outFilterScoreMin 10 
--outSAMattrRGline ID:$bnm 
--alignEndsType EndToEnd > $res$bnm$bam
```

I ran nearly identical commands (only changed the `--genomeSAindexNbases` parameter to 9.13) with the same set of reads on the same computer to the mouse RepBase by accident and it worked fine, but for some reason I'm getting a segfault now.

My computer is Ubuntu 16.10 with 6 physical (12 virtual) cores and 32GB of RAM. 
ETA: STAR is version 2.5.3a

Any ideas?

[Log.txt](https://github.com/alexdobin/STAR/files/1048570/Log.txt)

**Comment by JCSzamosi** [comment]
Update: I reran the indexing command with `--genomeSAindexNbases=3` as recommended elsewhere, but it did not solve the issue.

**Comment by JCSzamosi** [comment]
Further update: Despite the fact that there is a segmentation fault reported to the terminal (stderr?), the log files all report successful completion. This is very mysterious.

**Comment by alexdobin** [comment]
Hi @JCSzamosi 

it looks like this seg-fault happens after the run is completed - hence "ALL DONE" at the end of Log.out .
This is harmless since it happens when STAR exits and OS tries to free the used RAM.
This only happens on some systems and I thought I fixed it in 2.5.3a - but apparently not.
You can check for "ALL DONE" line in Log.out or for the presence of non-empty Log.final.out file - both indicate successful completion.

Cheers
Alex
